astronomer/providers/google/cloud/operators/bigquery.py

```diff
@@ -124,18 +124,23 @@ def execute(self, context: Context) -> None:  # noqa: D102
 
         self.job_id = job.job_id
         context["ti"].xcom_push(key="job_id", value=self.job_id)
-        self.defer(
-            timeout=self.execution_timeout,
-            trigger=BigQueryInsertJobTrigger(
-                conn_id=self.gcp_conn_id,
-                job_id=self.job_id,
-                project_id=self.project_id,
-                impersonation_chain=self.impersonation_chain,
-                poll_interval=self.poll_interval,
-                **kwargs,
-            ),
-            method_name="execute_complete",
-        )
+        if job.running():
```

Comment:
Can you please point me to the implementation of the running method?

Reply:
https://github.com/googleapis/python-api-core/blob/6acb3a95462f644037d4cf5f3c4d6a8a6149caa8/google/api_core/future/polling.py#L122

Reply:
which calls https://github.com/googleapis/python-api-core/blob/6acb3a95462f644037d4cf5f3c4d6a8a6149caa8/google/api_core/future/polling.py#L122

Reply:
yes, I was able to track it until there but where is https://github.com/googleapis/python-api-core/blob/6acb3a95462f644037d4cf5f3c4d6a8a6149caa8/google/api_core/future/polling.py#LL124C20-L124C31 (self.done() ) implemented

Reply:
https://github.com/googleapis/python-bigquery/blob/main/google/cloud/bigquery/job/base.py#L855

Sorry, I paste the wrong link in the previous comment